psychopy/experiment/components/button/__init__.py

```diff
@@ -0,0 +1,210 @@
+            updates='constant', allowedUpdates=_allow3[:],
+            hint=_translate("Defines the space between text and the textbox border"),
+            label=_localized['padding'])
+        self.params['anchor'] = Param(
+            anchor, valType='str', inputType="choice", categ='Layout',
+            allowedVals=['center',
+                         'top-center',
+                         'bottom-center',
+                         'center-left',
+                         'center-right',
+                         'top-left',
+                         'top-right',
+                         'bottom-left',
+                         'bottom-right',
+                         ],
+            updates='constant',
+            hint=_translate("Should text anchor to the top, center or bottom of the box?"),
+            label=_localized['anchor'])
+        self.params['borderWidth'] = Param(
+            borderWidth, valType='num', inputType="single", allowedTypes=[], categ='Appearance',
+            updates='constant', allowedUpdates=_allow3[:],
+            hint=_translate("Textbox border width"),
+            label=_localized['borderWidth'])
+
+
+    def writeInitCode(self, buff):
+        # do we need units code?
+        if self.params['units'].val == 'from exp settings':
+            unitsStr = ""
+        else:
+            unitsStr = "units=%(units)s," % self.params
+        # do writing of init
+        inits = getInitVals(self.params, 'PsychoPy')
+        code = (
+                       "%(name)s = visual.ButtonStim(win, \n"
+                       "    text=%(text)s, font=%(font)s,\n"
+                       "     pos=%(pos)s," + unitsStr + "\n"
+                       "     letterHeight=%(letterHeight)s,\n"
+                       "     size=%(size)s, borderWidth=%(borderWidth)s,\n"
+                       "     fillColor=%(fillColor)s, borderColor=%(borderColor)s,\n"
+                       "     color=%(color)s, colorSpace=%(colorSpace)s,\n"
+                       "     opacity=%(opacity)s,\n"
+                       "     bold=%(bold)s, italic=%(italic)s,\n"
+                       "     padding=%(padding)s,\n"
+                       "     anchor=%(anchor)s,\n"
+                       "     name='%(name)s')"
+        )
+        buff.writeIndentedLines(code % inits)
+
+    def writeFrameCode(self, buff):
+        BaseVisualComponent.writeFrameCode(self, buff)
+        # do writing of init
+        inits = getInitVals(self.params, 'PsychoPy')
+        callback = str(inits['callback'])
+        if inits['callback']:
+            # Add indent to any linebreaks to account for the loop
+            callback = str(inits['callback']).replace("\n", "\n   ")
+        else:
+            callback = "pass"
+        if inits['forceEndRoutine'].val:
+            endRt = "continueRoutine = False"
+        else:
+            endRt = ""
+        if inits['oncePerClick'].val:
+            hammingStr = "   while %(name)s.isClicked:\n      pass\n"
```

Comment:
hamming? For a sound that indicates the smoothing filter to prevent audio clicks. I haven't heard the term with button presses before